src/overlays/actors/ovl_En_Talk_Gibud/z_en_talk_gibud.c

```diff
@@ -28,166 +83,1057 @@ const ActorInit En_Talk_Gibud_InitVars = {
+                EnTalkGibud_SetupGrabFail(this);
+            } else if (globalCtx->grabPlayer(globalCtx, player)) {
+                EnTalkGibud_SetupGrab(this);
+            }
+        } else {
+            if (this->playerStunWaitTimer == 0) {
+                player->actor.freezeTimer = 40;
+                this->playerStunWaitTimer = 60;
+                func_8013ECE0(this->actor.xzDistToPlayer, 255, 20, 150);
+                func_80123E90(globalCtx, &this->actor);
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_REDEAD_AIM);
+            } else {
+                this->playerStunWaitTimer--;
+            }
+        }
+    } else if (this->grabWaitTimer == 0 && this->actor.xzDistToPlayer <= 45.0f) {
+        EnTalkGibud_SetupWalkToHome(this);
+    } else if (EnTalkGibud_PlayerOutOfRange(this, globalCtx)) {
+        EnTalkGibud_SetupWalkToHome(this);
+    }
+    if (this->grabWaitTimer > 0) {
+        this->grabWaitTimer--;
+    }
+
+    if (Animation_OnFrame(&this->skelAnime, 10.0f) || Animation_OnFrame(&this->skelAnime, 22.0f)) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_RIZA_WALK);
+    } else if (!(globalCtx->gameplayFrames & 0x5F)) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_REDEAD_CRY);
+    }
+}
+
+void EnTalkGibud_SetupGrab(EnTalkGibud* this) {
+    func_800BDC5C(&this->skelAnime, sAnimations, EN_TALK_GIBUD_ANIMATION_GRAB_START);
+    this->grabDamageTimer = 0;
+    this->actor.flags &= ~1;
+    this->grabState = EN_TALK_GIBUD_GRAB_START;
+    this->actionFunc = EnTalkGibud_Grab;
+}
+
+void EnTalkGibud_Grab(EnTalkGibud* this, GlobalContext* globalCtx) {
+    // This function needs to have two different temps for Player to match,
+    // but you don't have to necessarily use them both. This is just the most
+    // likely scenario; they got an Actor* pointer in the first temp, then
+    // casted it to Player* in the second temp.
+    Actor* playerActor = &GET_PLAYER(globalCtx)->actor;
+    Player* player = (Player*)playerActor;
+    s32 inPositionToAttack;
+    u16 damageSfxId;
+
+    switch (this->grabState) {
+        case EN_TALK_GIBUD_GRAB_START:
+            inPositionToAttack = EnTalkGibud_MoveToIdealGrabPositionAndRotation(this, globalCtx);
+            // Needs to have this "== true" to match.
+            if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame) && inPositionToAttack == true) {
+                this->grabState = EN_TALK_GIBUD_GRAB_ATTACK;
+                func_800BDC5C(&this->skelAnime, sAnimations, EN_TALK_GIBUD_ANIMATION_GRAB_ATTACK);
+            }
+            break;
+
+        case EN_TALK_GIBUD_GRAB_ATTACK:
+            if (this->grabDamageTimer == 20) {
+                s16 requiredScopeTemp;
 
-extern ColliderCylinderInit D_80B01350;
-extern DamageTable D_80B0137C;
-extern CollisionCheckInfoInit2 D_80B0139C;
-extern InitChainEntry D_80B01448[];
+                damageSfxId = player->ageProperties->unk_92 + NA_SE_VO_LI_DAMAGE_S;
```

Comment:
Now I'm really curious what they've done to replace `ageProperties` in MM. No actual suggestion, but do you know how large `unk_92` goes or if this is used just for Gibdos in the well?

Reply:
It's used for all forms of Redead/Gibdo (mainly because all three of them are heavy on the copypasta), as well as Floormasters. I checked out `unk_92` in-game using kz and it seemed to be 0x0000 for all forms of Link, including Fierce Diety. So I'm not exactly sure what it's doing here; perhaps `func_800B8E58` can tell us.